extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheMongoEntityBase.java

```diff
@@ -875,6 +875,55 @@ public static void persistOrUpdate(Object firstEntity, Object... entities) {
         MongoOperations.persistOrUpdate(firstEntity, entities);
     }
 
+    /**
+     * Update all entities of this type by the given update document, with optional indexed parameters.
```

Comment:
It's not really a `document` here, is it?

Reply:
Also, I read the whole PR and I'm not sure what syntax is allowed in the `update` param besides the mongo native query. It says this is a query string, but it's not, is it?

Can I do `foo = :name, bar = :lastname` or `gee = 2`?

Reply:
Yes, you can do it.

In reality, everything is a document in MongoDB, so there is no difference between what we called a query (MongoDB called it a filter document) and a stored document.

So you can use PanacheQL to express the update document, all the following are working update document:
```
FooBar.update("foo = :name and bar = :lastname", parameters).all()
FooBar.update("foo = :name, bar = :lastname", parameters).all()
FooBar.update("foo = ?1 and bar = ?2", name, lastname).all()
FooBar.update("foo = ?1, bar = ?2", name, lastname).all()
FooBar.update("{'foo' : :name, 'bar' : :lastname}", parameters).all()
FooBar.update("{'foo' : ?1, 'bar' : ?2}", name, lastname).all()
FooBar.update("foo", name).all()
```

Reply:
I added some sentence on the guide to document this inside the 'Simplified query' section.